### PR TITLE
[doc] Add more example how to use credentials

### PR DESCRIPTION
**Pull request by csordasmarton**
Add a note and a new example for the credentials sections which tells the user not to use protocols when adding new credentials entries.

### CODE REVIEW

File: docs/web/authentication.md [hunk 1]
```diff
@@ -362,6 +363,9 @@ Is it possible to generate a token from command line which can be used to
 authenticate in the name of the given user. This way no need to store passwords
 in text files. For more information [see](#personal-access-token).
 
+Note: do not give protocols in the credentials section. Use `my.company.org:443`
+instead of `https://my.company.org:443`.
```
Comment (bruntib) [inline docs/web/authentication.md]: what happens if I give a protocol? It is not really intuitive that the protocol part of a standard URL format is prohibited. Couldn't we cut it if it isn't necessary?

Comment (csordasmarton) [reply docs/web/authentication.md]: @gyorb already refactored this in #1460 pull request but it is still in work in progress status.

Comment (csordasmarton) [reply docs/web/authentication.md]: @bruntib I refactored this and I now cut the protocols and product names from the credentials.

File: web/client/codechecker_client/credential_manager.py [hunk 2]
```diff
@@ -27,6 +28,33 @@
...
 
 
+def simplify_credentials(credentials):
+    """ Replace the key of the entries with host:port values.
+
+    Returns a new dictionary from the credentials where the key will be
+    replaced by 'host:port' values so protocols and product names will be
+    removed if these are given.
+    """
+    host_entry_pattern = re.compile(
```
Comment (gyorb) [inline web/client/codechecker_client/credential_manager.py]: Wouldn't be easier to use the `urlparse.urlsplit` to split up the url?

Comment (csordasmarton) [reply web/client/codechecker_client/credential_manager.py]: I checked it and it works differently on different parameters and it is hard to decide which field we need to use and we also have to post-process the result.

```py
urlparse.urlsplit("my.server.com:443/Test")
# SplitResult(scheme='my.server.com', netloc='', path='443/Test', query='', fragment='')

urlparse.urlsplit("my.server.com:443")
SplitResult(scheme='', netloc='', path='my.server.com:443', query='', fragment='')

urlparse.urlsplit("*:8080")
# SplitResult(scheme='', netloc='', path='*:8080', query='', fragment='')

urlparse.urlsplit("https://my.server.com:443/Test")
# SplitResult(scheme='https', netloc='my.server.com:443', path='/Test', query='', fragment='')
```

**Pull request merged**